lib/Sources/App/SearchBarModel.swift

```diff
@@ -46,8 +49,14 @@ public final class SearchBarModel {
     func didSubmitSearchString() {
         guard let sanitized = sanitizeInput(query: searchString) else { return }
         self.focusedField = nil
+
+        // If the user typed a full URL
         if let derivedUrl = URL(string: sanitized), derivedUrl.isHttp {
             onSubmit(derivedUrl)
+        // If the user typed something that resolves to a host. i.e. www.google.com or amazon.co.uk
+        } else if let hostnameRegex, (try? hostnameRegex.wholeMatch(in: sanitized)) != nil, gethostbyname(sanitized) != nil, let url = URL(string: "https://" + sanitized) {
```

Comment:
This is a bit dense here within the existing conditional chain, would be nice for readability to break out the logic e.g.
```suggestion
        } else if let url = hostNameUrl(hostname: sanitized) {
```

And a convenience function something like:
```swift
    private func hostNameUrl(hostname: String) -> URL? {
        guard (try? hostnameRegex?.wholeMatch(in: hostname)) != nil else {
            return nil
        }
        guard gethostbyname(sanitized) != nil else {
            return nil
        }
        return URL(string: "https://" + sanitized)
    }
```